mm2src/coins/eth/eth_balance_events.rs

```diff
@@ -80,31 +120,69 @@ impl EventBehaviour for EthCoin {
+                        ctx.stream_channel_controller
+                            .broadcast(Event::new(
+                                format!("{}:{}", EthCoin::ERROR_EVENT_NAME, coin.ticker),
+                                e.to_string(),
+                            ))
+                            .await;
+                        sleep_remaining_time(interval, now).await;
+                        continue;
+                    },
+                };
+
                 let mut balance_updates = vec![];
-                for result in get_all_balance_results_concurrently(&coin).await {
+                for result in get_all_balance_results_concurrently(&coin, addresses).await {
                     match result {
-                        Ok((ticker, balance)) => {
-                            if Some(&balance) == cache.get(&ticker) {
+                        Ok(res) => {
+                            if Some(&res.balance) == cache.get(&res.ticker).and_then(|map| map.get(&res.address)) {
                                 continue;
                             }
 
                             balance_updates.push(json!({
-                                "ticker": ticker,
-                                "balance": { "spendable": balance, "unspendable": BigDecimal::default() }
+                                "ticker": res.ticker,
+                                "address": res.address,
+                                "balance": { "spendable": res.balance, "unspendable": BigDecimal::default() }
                             }));
-                            cache.insert(ticker.to_owned(), balance);
+                            cache
+                                .entry(res.ticker.clone())
+                                .or_insert_with(HashMap::new)
+                                .insert(res.address, res.balance);
                         },
-                        Err((ticker, e)) => {
-                            log::error!("Failed getting balance for '{ticker}' with {interval} interval. Error: {e}");
-                            let e = serde_json::to_value(e).expect("Serialization should't fail.");
+                        Err(err) => {
+                            log::error!(
+                                "Failed getting balance for '{}:{}' with {interval} interval. Error: {}",
+                                err.ticker,
+                                err.address,
+                                err.error
+                            );
+                            let e = serde_json::to_value(err.error).expect("Serialization shouldn't fail.");
                             ctx.stream_channel_controller
                                 .broadcast(Event::new(
-                                    format!("{}:{}", EthCoin::ERROR_EVENT_NAME, ticker),
+                                    format!("{}:{}:{}", EthCoin::ERROR_EVENT_NAME, err.ticker, err.address),
```

Comment:
Wondering why we did not create a enum for event names as a common registry? As I can see, we duplicate event names in different coins. Overlaps are also possible if no single registry

Reply:
Since we started with an implementation for one coin and added other coins gradually I guess this wasn't considered, but it was considered in a recent review comment ref. https://github.com/KomodoPlatform/komodo-defi-framework/pull/2076#discussion_r1537603729
I can refactor it here but need to merge with latest dev first
c.c. @onur-ozkan

Reply:
Fixed here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1979/commits/2d8b0e2cb9e4c8a7b438cce1f87d433e4caec2c3
c.c. @onur-ozkan @borngraced

Reply:
The format is good (I am using similar format for another event here [1](https://github.com/KomodoPlatform/komodo-defi-framework/blob/985997b47d84f9025e371be0f7e971e6bcdb0be5/mm2src/mm2_core/src/data_asker.rs#L72) [2](https://github.com/KomodoPlatform/komodo-defi-framework/blob/985997b47d84f9025e371be0f7e971e6bcdb0be5/mm2src/coins/tendermint/tendermint_coin.rs#L1051)), but we need a format consistency for per events (if we use this format, we should use it on all other `COIN_BALANCE` messages too.), otherwise it would be painful from clients to parse the event.